### ISSUE TITLE
WrappedVulkan::Serialise_InitialState() attempts to use vkCmdCopyBufferToImage() to initialize multisample image data

### ISSUE THREAD

**Issue by kbiElude** [baldurk/renderdoc#193]
If an app creates a MS image, RenderDoc will try to re-create its contents by doing buffer->image copies for all relevant subresource ranges. There are two problems here:
- vkCmdCopyBufferToImage() does not support MS images.
- The function tries to create a MS image with VK_IMAGE_USAGE_TRANSFER_SRC_BIT and VK_IMAGE_USAGE_TRANSFER_DST_BIT. At least the latter is invalid, according to this excerpt from the spec:

```
VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL: must only be used as a destination image of a transfer command. This layout is valid only for subresources of images created with the VK_IMAGE_USAGE_TRANSFER_DST_BIT usage bit enabled.
```

**Comment by baldurk** [comment]
I thought I had a TODO lying around somewhere for handling multisampled images, but I guess not. You're right though, I need a specific path for handling it.

I'm not sure I understand why that spec line is saying VK_IMAGE_USAGE_TRANSFER_SRC_BIT or VK_IMAGE_USAGE_TRANSFER_DST_BIT is invalid for MSAA images (since it would be needed for normal vkCmdCopyImageToImage which should work), but I won't be using transfers anyway. The way I handled it on the other APIs was to run a compute shader that did a copy between an MSAA image and an array image with `# slices = # samples`. That should be doable on Vulkan too I just haven't implemented it yet.

**Comment by kbiElude** [comment]
You're right about the image->image copy ops - I just checked the spec and it turns out I was wrong. Still, that MS image support would be a nice-to-have :)

The CS-based approach is the right thing to do here.

**Comment by baldurk** [comment]
I believe that commit should fix the issues you were seeing, it adds full support for saving/loading multisampled initial states.

**Comment by baldurk** [comment]
This should be resolved in v0.32. If something's still not right in the latest build then let me know and I'll re-open and investigate.
